lib/utils.coffee

```diff
@@ -40,17 +40,23 @@ exports.TOKEN_REFRESH_INTERVAL = 1 * 1000 * 60 * 60 # 1 hour in milliseconds
 # This function makes use of a soft user-configurable setting called `tokenRefreshInterval`.
 # That setting doesn't express that the token is "invalid", but represents that it is a good time for the token to be updated *before* it get's outdated.
 #
-# @param {Object} tokenInstance - an instance of `resin-token`
+# @param {Object} tokenInstance - an instance of `resin-auth`
 # @returns {Promise<Boolean>} the token should be updated
 #
 # @example
-# tokenUtils.shouldUpdateToken(tokenInstance).then (shouldUpdateToken) ->
-#		if shouldUpdateToken
+# tokenUtils.shouldRefreshKey(tokenInstance).then (shouldRefreshKey) ->
+#		if shouldRefreshKey
 #			console.log('Updating token!')
 ###
-exports.shouldUpdateToken = (token) ->
-	token.getAge().then (age) ->
-		return age >= exports.TOKEN_REFRESH_INTERVAL
+exports.shouldRefreshKey = (auth) ->
+	auth.hasKey().then (hasToken) ->
+		if !hasToken
+			return false
+		auth.getType().then (type) ->
+			if type != 'JWT'
+				return false
```

Comment:
This doesn't do anything. You've created a promise here that resolves to undefined (for an api key) or false (for a token), but then nobody waits for or looks at the result. The getAge line below runs in parallel to this - it's not waiting for or using this result at all. Does that make sense? I'm not sure how familiar with promises you are - let me know if you want me to explain that in more depth.

Reply:
Sorry, I missed the indentation.